### PR TITLE
Feat: Add preliminary workflow for end-to-end testing of package

### PR DESCRIPTION
**Pull request by vibalre**
### Changes
Related #58 

* Adds github action workflow for running main Rhino functions (except `test_r()`)
* Adds template scripts for `build_sass()`, `build_js()`, and `format_r()`

### Workflow Example

* See results of [example successful pipeline](https://github.com/Appsilon/rhino/actions/runs/4980832156)

### CODE REVIEW

File: tests/end2end/test-format-r.R [hunk 1]
```diff
@@ -0,0 +1,3 @@
+paths <- commandArgs(trailingOnly = TRUE)
```
Comment (vibalre) [inline tests/end2end/test-format-r.R:1]: Similar to `shiny.benchmark`, we can use this script, if ever we need to expand the tests on `format_r()` to check the results of different directories.

**Review by vibalre — COMMENTED** [review]
A very minimal workflow for Rhino end-to-end test. All functions worked by changing the working directory to the create Rhino application. However, this does not work for `test_r()` as it requires changing the working directory and restarting the R session. I have not figured out how to do that yet via command line.

File: .github/workflows/e2e-test.yml [hunk 2]
```diff
@@ -0,0 +1,107 @@
+on: push
+
+name: Rhino Usage Test
```
Comment (kamilzyla) [inline .github/workflows/e2e-test.yml:3]: Let's stick to calling this "E2E Test".

```suggestion
name: E2E Test
```

File: .github/workflows/e2e-test.yml [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+        config:
+        - {os: macOS-latest,   r: 'release'}
+        - {os: windows-latest, r: 'release'}
+        - {os: ubuntu-22.04, r: 'devel'}
+        - {os: ubuntu-22.04, r: 'release'}
+        - {os: ubuntu-22.04, r: 'oldrel'}
+
+    env:
+      GITHUB_PAT: ${{ secrets.GITHUB_TOKEN }}
+      BRANCH_NAME: ${{ github.head_ref || github.ref_name }}
```
Comment (kamilzyla) [inline .github/workflows/e2e-test.yml:29]: Are this used or necessary?

```suggestion
```

File: .github/workflows/e2e-test.yml [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+
+      - name: Install R
+        uses: r-lib/actions/setup-r@v2
+        with:
+          r-version: ${{ matrix.config.r }}
+
+      - name: Install R package dependencies
+        uses: r-lib/actions/setup-r-dependencies@v2
+        with:
+          extra-packages: local::. # Necessary to avoid object usage linter errors.
```
Comment (kamilzyla) [inline .github/workflows/e2e-test.yml:43]: This comment doesn't make sense in this workflow. In `ci.yml`, "object usage errors" will appear when running `lintr::lint_package()` if the package is not installed.
```suggestion
          extra-packages: local::.
```

**Review by kamilzyla — APPROVED** [review]
Thanks Recle! This is a huge step forward in terms of testing Rhino :100: :tada:

Consider editing your commit history (using interactive git rebase) to squash and organize things a bit :wink: A long, chronological log of tiny edits is rarely useful; fewer commits with meaningful messages can make the history more understandable and it can simplify things when merge conflicts arise.

File: .github/workflows/e2e-test.yml [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+
+      - name: Setup Node
+        uses: actions/setup-node@v3
```
[the remark this answers is not in the document]
Comment (vibalre) [reply .github/workflows/e2e-test.yml:42]: When I used `node-version: 18`, cypress timeouts. Not sure why, so I think we should keep 16 here for now.

File: tests/end2end/test-format-r.R (hunk 1 above)
Comment (vibalre) [reply tests/end2end/test-format-r.R]: I specified default paths to `app` and `tests`.

File: .github/workflows/e2e-test.yml (hunk 3 above)
Comment (vibalre) [reply .github/workflows/e2e-test.yml]: I kept this without checking. You are correct, these are not being used. I removed them.

File: .github/workflows/e2e-test.yml [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+        with:
+          node-version: 16
+
+      - name: Install rhino
+        if: always()
+        shell: Rscript {0}
+        run: |
+          install.packages('remotes')
+          remotes::install_local('../../')
```
[the remark this answers is not in the document]
Comment (vibalre) [reply .github/workflows/e2e-test.yml]: I did not understood the purpose of `extra-packages: local::.`. Yes, this step was unnecessary. Removed.

**Pull request merged**